docs/DeveloperGuide.md

```diff
@@ -1,26 +1,237 @@
+4.6 [Sort events](#46-sort-events) <br>
+4.7 [View events](#47-view-events) <br>
+5. [Documentation](#5-documentation) <br>
+5.1 [Product scope](#51-product-scope) <br>
+5.2 [User stories](#52-user-stories) <br>
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## 1. Introduction
+NUSchedule is a Command-Line based application that manages all of your commitments. Built with a clean and intuitive
+interface, NUSchedule will help you manage your commitments without any hassle. If you are a proficient in typing, 
+NUSchedule will prove itself to be the most efficient way to manage your commitments.  
+  
+This developer guide provides information on the architecture and design of the application, NUSchedule. This guide
+provides information that will not only help you get started as a NUSchedule contributor, but that you will find useful
+to refer to even if you are already a contributor
 
+## 2. Getting Started
+This section provides information to help you get NUSchedule up and running on your own computers.
 
-## Product scope
-### Target user profile
+### 2.1 Prerequisites
+1. JDK 11
+2. Intellij IDEA
 
-{Describe the target user profile}
+### 2.2 Setting Up
+1. Fork [this repo](https://github.com/AY2021S1-CS2113T-F14-4/tp), and clone the fork onto your computer.   
+2. Open IntelliJ (if you are not in the welcome screen, click File > Close Project
+to close the existing project dialog 
+first).   
+3. Set up the correct JDK version for Gradle  
+    1. Click Configure > Project Defaults > Project Structure
+    2. Click New… and set it to the directory of the JDK.
+4. Click Import Project (or Open or Import in newer version of Intellij).
+5. Locate the build.gradle file (not the root folder as you would do in a normal importing) and select it. Click OK.
+   If asked, choose to Open as Project (not Open as File).
+6. Click OK to accept the default settings but do ensure that the selected version of Gradle JVM matches the JDK 
+   being used for the project.
 
-### Value proposition
+## 3. Design
+This section describes the different components of the application and how each component interact with
+each other to run the program. 
+
+### 3.1 Architecture
+![architecture](../diagrams/architecture.png)
+
+The Architecture Diagram above provides a high-level view of the design of NUSchedule. The app can be broken down into 
+5 different components:  
+1. Main: Initializes the other components and connects them with each other.
+2. UI: Manages the User Interface that the user interacts with.
+3. Logic: Interprets user commands.
+4. Storage: Reads data from and writes data to the hard disk.
+5. Model: Stores the data the app uses in memory.
+
+__How the architecture components interact with each other__  
+The sequence diagram below shows how each individual component interacts with each other when the user inputs a command.
+![sequence](../diagrams/ArchitectureSequence.png)
```

Comment:
Unable to load architecture and sequence images.

![image](https://user-images.githubusercontent.com/8433829/97529796-a4556180-19eb-11eb-9b7c-ec9d934f1687.png)